main.go

```diff
@@ -23,6 +24,10 @@ func main() {
 	serverConf := server.Config{
 		HTTPListenAddr: conf.Server.Address,
 		HTTPListenPort: conf.Server.Port,
+		HTTPMiddleware: []middleware.Interface{
+			gateway.NewAuthentication(&conf),
+		},
+		UnAuthorizedHTTPListenPort: 6666,
```

Comment:
The name `Admin` seems fine. I could not come up with a name so I hard-coded the port number. However, I designed `server` struct for this change. So, it will be easy to change.